### PR TITLE
Admin: Customer List (#135)

### LINKED ISSUE #135: Admin: Customer List
## User Story

> As a **WDI Staff (Admin)**,
> I need to view a list of registered customers and their profile details,
> in order to find and review customer information quickly.

## Background

This is the read-only view of customer data in the admin area. Editing actions (email, password, admin rights) are covered in the Admin: Customer Management story. The customer list is the entry point for all customer-related admin work.

## Scope

**In scope:**
- Paginated list of all registered customers showing first name, last name, email address, and IsAdmin (Yes/blank)
- List sorted by last name
- Compact table layout to minimize vertical scrolling
- Pagination controls: First, Previous, Next, Last using Font Awesome icons; 20 customers per page
- CSS reference to Font Awesome CDN (free version) added to the app shell
- Empty state when no customers are registered
- Read-only profile view for an individual customer showing their full profile details
- Profile view indicates whether the customer has admin rights

**Out of scope:**
- Search or filtering the customer list (deferred to a later milestone)
- Any editing actions (covered in story 03-admin-customer-management)

## Developer Notes

- The list and profile view live under `Features/Admin/` and use the admin layout.
- Pagination controls use Font Awesome free icons (e.g. `fa-angles-left`, `fa-angle-left`, `fa-angle-right`, `fa-angles-right`) instead of text labels. Add a `<link>` to the Font Awesome free CDN in `App.razor` (or the admin layout) — do not bundle or self-host.

### Testing Pagination

A local dev database rarely has 21+ customers, so two complementary strategies are recommended:

- **Unit tests (Option A):** Use the existing `CreateDb()` + seed pattern in `WidgetDepot.Tests` to seed 25 customers in-memory and assert correct Skip/Take math, boundary conditions (0, 20, 21 rows), and page-count calculation. This is the primary correctness check and requires no extra infrastructure.

- **Configurable page size for E2E (Option C):** Add a `Pagination:PageSize` key to `appsettings.json` (default `20`) in the Web project and pass it to the list query endpoint/handler in the ApiService. In `playwright.config.ts`, pass `--Pagination:PageSize=3` to the `webServer` command so the E2E suite only needs 5–6 customers (achievable with the existing registration flow) to exercise all four pagination buttons.

## Acceptance Criteria

- [x] Admin can navigate to the customer list from the admin navigation
- [x] The list displays each customer's first name, last name, email address, and IsAdmin flag
- [x] The list is sorted by last name
- [x] The list is paginated with 20 customers per page
- [x] Pagination controls display Font Awesome icons for First, Previous, Next, and Last (no text labels)
- [x] Font Awesome free CDN stylesheet is referenced in the app shell
- [x] The table uses a compact layout to minimize vertical scrolling
- [x] If there are no registered customers, the list shows an empty state message
- [x] Clicking a customer opens a read-only profile view showing their full profile details
- [x] The profile view includes a visible indicator if the customer has admin rights

## Refinement Notes

Search and filtering are explicitly deferred to a later milestone — confirmed with user. Pagination style (First/Prev/Next/Last, 20 per page) and compact table layout specified by the user to reduce scrolling on busy lists.

### PR DESCRIPTION
**Pull request by edgamat**
## Summary

- Adds paginated, read-only admin customer list at `/admin/customers` showing last name, first name, email, and admin flag — sorted by last name, 20 per page
- Adds read-only customer profile page at `/admin/customers/{id}` with full details and an Admin badge for admin users
- Adds Font Awesome free CDN to `App.razor` for pagination icon support
- Updates admin nav link from *Users* → *Customers*
- Adds configurable `Pagination:PageSize` (default 20) so E2E tests can override to a small value

## Test plan

- [x] 15 new unit tests covering `GetCustomerListHandler` (pagination math, boundary conditions, sort order, IsAdmin flag) and `GetCustomerProfileHandler` (profile fields, addresses, not-found)
- [x] 307 unit tests pass (`dotnet test`)
- [x] E2E page objects (`CustomerListPage`, `CustomerProfilePage`) added with POM pattern; 9 new E2E scenarios in `customers.spec.ts`
- [x] E2E TypeScript parses cleanly (`npx playwright test --list` — 41 tests in 7 files)
- [x] `playwright.config.ts` sets `Pagination__PageSize=3` so pagination buttons can be exercised with a handful of registered customers

Closes #135

### CODE REVIEW

File: src/WidgetDepot.ApiService/Features/Admin/Customers/CustomerEndpointExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+namespace WidgetDepot.ApiService.Features.Admin.Customers;
+
+public static class CustomerEndpointExtensions
```
Comment (edgamat) [inline src/WidgetDepot.ApiService/Features/Admin/Customers/CustomerEndpointExtensions.cs:6]: All the other features are organized using a separate sub-folder for each use case, and separate endpoint files. See the orders features as an example: src/WidgetDepot.ApiService/Features/Orders

Please reorganize the Customer folder to follow the same pattern.

File: tests/WidgetDepot.E2E/tests/customers.spec.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    await customerProfilePage.goto(1);
+    await customerProfilePage.waitForReady();
+
+    const isNotFound = await customerProfilePage.notFoundAlert.isVisible().catch(() => false);
+    if (isNotFound) {
+      test.skip();
+      return;
+    }
+
+    await expect(customerProfilePage.adminBadge).toBeVisible();
```
Comment (edgamat) [inline tests/WidgetDepot.E2E/tests/customers.spec.ts:116]: This assertion fails because there are 2 elements on the page that match the text "Admin". Please change the page object model to be more specific when selecting the admin badge.
